### PR TITLE
Fuse group convolution

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: src/simplify_algebra.cpp [hunk 1]
```diff
@@ -725,6 +725,54 @@ struct find_concat_op
...
+        return match::name("concat")(
+            match::all_of[match::inputs()](match::used_once(), match::name("convolution")));
+    }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto ins  = r.result;
+        auto axis = ins->get_operator().to_value()["axis"].to<int>();
+        if(axis != 1)
+            return;
```
Comment (TedThemistokleous) [inline src/simplify_algebra.cpp:741]: Add test case to cover this return path.

**Review by TedThemistokleous — CHANGES_REQUESTED** [review]
Just add a test case to cover the axis != case in your apply. Otherwise LGTM

File: src/simplify_algebra.cpp [hunk 2]
```diff
@@ -725,6 +725,54 @@ struct find_concat_op
...
+        if(ins->inputs().empty())
+            return;
```
Comment (umangyadav) [inline src/simplify_algebra.cpp:743]: Would this case occure at all ? All of the inputs to the concat must be convolution. It can't be empty

**Review by CharlieL7 — APPROVED** [review]
LGTM

**Review by TedThemistokleous — APPROVED** [review]
Just fix licensing now. LGTM too

**Pull request merged**